Minor clean up

bmarika opened the pull request.

Commit:
Remove reset migrations; It is nolonger needed

Commit:
Stop droping and creating the db during setup.

Commit:
Warm up the facilities list endpoint

Commit:
Fix failing test

ngash commented on fabfile.py:
```diff
@@ -103,18 +92,6 @@ def setup(*args, **kwargs):
     # needs to come first to as to index data as it is being loaded
     remove_search_index()
     create_search_index()
-    no_sudo = True if 'no-sudo' in args else False
```
was this supposed to be removed or just to be done outside `fab setup` ?

bmarika replied:
You have a point. A function with the db stuff would be better here

ngash commented on fabfile.py:
```diff
@@ -44,17 +44,6 @@ def server_deploy():
-    for app_name in base.LOCAL_APPS:
-        manage('makemigrations {}'.format(app_name))
-
-    local('git add . --all')
```
hii ilikuwa ukora

ngurenyaga replied:
Guilty as charged. It had its place and time - when it was me and @bmarika only working in the same room, shouting changes across the table.

That time is gone.

Commit:
Bring out creation of the database as a separate function.
Completely remove fab setup and leave individual separate functions to undertake its purpose.

Commit:
Add recreate the search index.

Commit:
Clean up

Commit:
Update playbooks

Commit:
Create new db only when needed

Commit:
Update docs

ngash commented on docs/02_developer_install.rst:
```diff
@@ -48,19 +48,19 @@ Getting started
-    * Run migrations.
-
-    * Load demo data if the project **DEBUG** attribute in settings is set to true .
+3. Run the following commands sequentially:
```
Separate the setup install commands into:
1) Setup with existing database
2) Setup without existing database 
3) Recreating the database

Commit:
Add setup db alone and recreating the db to the docs

Commit:
Add more emphasis on the retention and recreation of the database
